watershed_workflow/condition.py

```diff
@@ -392,5 +391,185 @@ def fill_gaps(img_in, nodata=np.nan):
 
     # interpolate the whole image
     return interp0(np.ravel(xx), np.ravel(yy)).reshape(xx.shape)
+    
+
+def condition_river_mesh(m2, river, smooth=False, use_parent=False, lower=False, use_nhd_elev=False, treat_banks=False, depress_by=0):
+    """Conditoning the elevations of stream-corridor elements (generally required in flat agricultural watersheds) to ensure connectivity throgh culverts, 
+    skips ponds, maintaiin monotonicity, enforce depths of constructed channels
+
+    Parameters:
+    -----------
+    
+    m2: watershed_workflow.mesh.Mesh2D object
+        2D mesh elevated on DEMs
+    river: watershed_workflow.river_tree.River object
+        river tree with node.elements added for quads
+    smooth: boolean, optional
+        flag for smoothing the profile of each reach using a gaussian filter (mainly to pass through railroads and avoid reservoirs)
+    use_parent: boolean, optional
+        if to use segment of parent node while smoothing (seems to be not making a huge difference)
+    lower: boolean, optional
+        to lower the smoothed bed profile to match the lower points on the raw bed profile. This is useful particularly for narrow ag. dicthes
```

Comment:
spelling/typos:

 dicthes --> ditches, 
sream --> stream, 
reached --> reaches,
"we banks may fall" --> ???